server/services/log/file/file.go

```diff
@@ -0,0 +1,79 @@
+package file
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"go.woodpecker-ci.org/woodpecker/v2/server/model"
+	"go.woodpecker-ci.org/woodpecker/v2/server/services/log"
+)
+
+type logStore struct {
+	base string
+}
+
+func NewLogStore(base string) (log.Service, error) {
+	if base == "" {
+		return nil, fmt.Errorf("file storage base path is required")
+	}
+	if _, err := os.Stat(base); err != nil && os.IsNotExist(err) {
+		err = os.MkdirAll(base, 0600)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return logStore{base: base}, nil
+}
+
+func (l logStore) filePath(id int64) string {
+	return filepath.Join(l.base, strconv.Itoa(int(id))+".json")
```

Comment:
Would it make sense to include the repo id in the path to improve grouping in the folder?